src/pagination/renderer/ScrollBulletRenderer.ts

```diff
@@ -0,0 +1,129 @@
+import { DIRECTION, FlickingError } from "@egjs/flicking";
+
+import { PAGINATION } from "../../const";
+import { BROWSER } from "../../event";
+import { addClass, removeClass } from "../../utils";
+
+import Renderer from "./Renderer";
+
+class ScrollBulletRenderer extends Renderer {
+  private _bullets: HTMLElement[] = [];
+  private _bulletSize: number = 0;
+  private _previousIndex: number = -1;
+  private _sliderIndex: number = -1;
+
+  public render() {
+    const wrapper = this._wrapper;
+    const flicking = this._flicking;
+    const pagination = this._pagination;
+    const renderBullet = pagination.renderBullet;
+    const anchorPoints = flicking.camera.anchorPoints;
+
+    const dynamicWrapperClass = `${pagination.classPrefix}-${PAGINATION.SCROLL_WRAPPER_SUFFIX}`;
+    const bulletClass = `${pagination.classPrefix}-${PAGINATION.BULLET_SUFFIX}`;
+    const sliderClass = `${pagination.classPrefix}-${PAGINATION.SCROLL_SLIDER_SUFFIX}`;
+    const uninitClass = `${pagination.classPrefix}-${PAGINATION.SCROLL_UNINIT_SUFFIX}`;
+
+    const sliderEl = document.createElement("div");
+
+    addClass(sliderEl, sliderClass);
+    addClass(wrapper, uninitClass);
+    addClass(wrapper, dynamicWrapperClass);
+    wrapper.appendChild(sliderEl);
+
+    sliderEl.innerHTML = anchorPoints
+      .map((_, index) => renderBullet(bulletClass, index))
+      .join("\n");
+
+    const bullets = [].slice.call(sliderEl.children) as HTMLElement[];
+
+    bullets.forEach((bullet, index) => {
+      const anchorPoint = anchorPoints[index];
+
+      bullet.addEventListener(BROWSER.MOUSE_DOWN, e => {
+        e.stopPropagation();
+      });
+
+      bullet.addEventListener(BROWSER.TOUCH_START, e => {
+        e.stopPropagation();
+      });
+
+      bullet.addEventListener(BROWSER.CLICK, () => {
+        flicking.moveTo(anchorPoint.panel.index)
+          .catch(err => {
+            if (err instanceof FlickingError) return;
+            throw err;
+          });
+      });
+    });
+
+    if (bullets.length <= 0) return;
+
+    const bulletStyle = getComputedStyle(bullets[0]);
+    const bulletSize = bullets[0].clientWidth + parseFloat(bulletStyle.marginLeft) + parseFloat(bulletStyle.marginRight);
```

Comment:
parseFloat has [this problem](https://stackoverflow.com/questions/30367133/javascript-parsefloat-issue). Could you check it please?

Reply:
Well I think that's a common error case when using floating value, why do you think that can be problematic in this case?

Reply:
This can be a problem if `e` is included in the number.

But I don't think that will happen.

Reply:
![image](https://user-images.githubusercontent.com/26213435/123614657-7dcc1c80-d83f-11eb-8a2f-92d2ca4d9fc7.png)
I guess it's okay?